Add shell.nix

BlueGone opened the pull request:
This PR give an easy way to get a working environnement on this project without polluting his system-side installation with uneeded packages (sdl, miniupnpc, ...).
Insteed of installing all the prerequesites, the only requirement is having `nix` installed. Then, in s25client, just open `nix-shell`. This works on both Linux and Darwin/MacOSX.

Flamefire commented:
This is to set up a **development** environment, isn't it? (I'm not familiar with nix)

If so I suggest to add SDL2 and its mixer to the list as we will switch to SDL2 as the default soon. 

Besides that: I assume you tested this? Could you add step-by-step instructions to the Readme.MD so new users know about this? Looks like it will be very useful

BlueGone commented:
Yes, Nix gives a development environment. When opening a nix-shell in the project directory, a custom `PATH` env variable will be set, with all dependencies in it.

I just added SDL2 in the list of dependencies, and updated README.md to explain how to build for nix users.

The only test I made is building the whole project on a system with nothing installed (no cmake, SDL, not even curl), but nix.
